packages/block-editor/src/components/link-control/index.js

```diff
@@ -94,12 +78,8 @@ function LinkControl( {
 
 		// Populate input searcher whether
 		// the current link has a title.
-		if ( currentLink && currentLink.title ) {
-			setInputValue( currentLink.title );
-		}
-
-		if ( isFunction( onChangeMode ) ) {
-			onChangeMode( mode );
```

Comment:
Are we removing a public API here? If so I'm not clear on what this might effect but I think it was added for a reason (likely Nav Block related).

Reply:
I removed it because I didn't see any usage in our codebase. So I prefer for the "modes" to be considered implementation detail.

Reply:
There's been some churn here, but considering the primary usefulness in treating `mode` as a string might have been allow an external consumer to `switch` off `onChangeMode` (based on its introduction in #18225), the removal of the prop here would have allowed to simplify back to just managing this as a boolean of the `isEditingLink` state.

Reply:
> There's been some churn here, but considering the primary usefulness in treating `mode` as a string might have been allow an external consumer to `switch` off `onChangeMode` (based on its introduction in #18225), the removal of the prop here would have allowed to simplify back to just managing this as a boolean of the `isEditingLink` state.

See: #19737